enhancements/microshift/audit-log-configuration-options.md

```diff
@@ -0,0 +1,255 @@
+
+## Proposal
+
+This proposes exposing a subset of kube-apiserver audit-log settings flags. The kube-apiserver settings will enable user control over log file rotation and retention. Users may set fields in combination to define a maximum storage limit (e.g. max num files * max single file size = total storage limit). This is a critical feature for far edge devices with limited storage capacities. On such devices, logging data accumulation risks starving the host system or cluster workloads, potentially bricking the device until human intervention can be applied.  Thus, it is necessary to provide users a means of enforcing such a limit and what actions to take at that limit. Users must also be able to select which events are logged. This will be exposed as a set of "profiles," with predefined behaviours and will be mutually exclusive. This should give users holistic control of their audit log rotation, retention, and max allowable storage allocation.
+
+### Workflow Description
+
+1. Administrator edits MicroShift config file to specify desired audit logging policy profile
+2. Administrator edits MicroShift config file to specify max file size, number of files total, and max age of files for logs
+3. Administrator restarts MicroShift service to apply changes
+
+### API Extensions
+
+**Audit Log Policy Configuration:**
+
+- MicroShift config’s `apiServer` root-field must include a child node for storing log configurations.
+
+```yaml
+apiServer:
+  auditLog: MAP(STRING)INTERFACE
+```
+
+- The `apiServer.auditLog` field must include a child node for storing the audit log policy profile to use. See [Implementation Details](#implementation-detailsnotesconstraints) for more. These are the same policies as in [Openshift Audit log policies](https://docs.openshift.com/container-platform/4.14/security/audit-log-policy-config.html#about-audit-log-profiles_audit-log-policy-config).
+
+```yaml
+apiServer:
+  auditLog:
+    profile: STRING
+```
+
+**Audit Log File Rotation:**
+
+Log rotation is governed by 3 values defining file size, number of retained files, and the maximum age of retained files.  The values are processed independently of one another, without prioritization.  Max size and max number of files may be used in conjunction to limit the storage footprint of retained logs (e.g. `maxFileSize` * `maxFiles` = log storage upper limit)
+
+- `maxFileSize` specifies maximum audit log file size in megabytes.
+- `maxFiles` specifies the maximum number of rotated audit log files to retain.  Once this limit is reached, the apiserver will delete log files in order from oldest to newest, until all specified limits are satisfied. 
+  - _For example_,`maxFiles: 1` will result in only 1 file of size `maxFileSize` being retained in addition to the current active log, provided it also is within the `maxFileAge` limit, if specified.
+- `maxFileAge` specifies the maximum time in days to retain log files.  Files older than this limit will be deleted.
```

Comment:
I think these options aren't exposed in OCP, so we're going to need to document aspects of this that OCP does not. For example, which option takes precedence, age or space?

If I set maxAge to 1 and maxFiles to 1000 and I'm only generating 1 file per day, am I going to only have 1 file retained?

Reply:
That's correct. The fields are independently processed. That's mentioned on L78, but might need some additional info